### ISSUE TITLE
Point selection doesn't work at high zoom?

### ISSUE THREAD

**Issue by psobolewskiPhD** [napari/napari#3610]
## 🐛 Bug

When **really** zoomed in—pixel level—the point selection tool doesn't always select points.

https://user-images.githubusercontent.com/76622105/141532392-27a10d87-4e7b-4813-b5a5-c18e89e49daf.mp4

## To Reproduce

1. Open Astronaut sample
2. Add points layer
3. Make a few points
4. Zoom in on the points, select one and move it.
5. Repeat 4, zoom in, use Select tool to move points.

## Expected behavior

Points should "light up" on hover and be selectable at high zooms, just as at low zoom.

## Environment

napari: 0.4.12
Platform: macOS-12.0.1-arm64-arm-64bit
System: MacOS 12.0.1
Python: 3.9.6 | packaged by conda-forge | (default, Jul 6 2021, 08:51:19) [Clang 11.1.0 ]
Qt: 5.15.2
PyQt5: 5.15.4
NumPy: 1.21.0
SciPy: 1.7.0
Dask: 2021.06.2
VisPy: 0.7.0

OpenGL:
- GL version: 2.1 Metal - 76.1
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1680x1050, scale 2.0

Plugins:
- Measurements (skimage regionprops): 0.1.10
- StarDist: 2021.10.25
- Utilities (skimage regionprops): 0.1.10
- aicsimageio-in-memory: 0.4.1
- aicsimageio-out-of-memory: 0.4.1
- animation: 0.0.2
- clEsperanto: 0.10.7
- console: 0.0.4
- napari-tools-menu: 0.1.3
- ome-types: 0.2.6
- scikit-image: 0.4.12
- svg: 0.1.5

## Additional context

I noticed this same behavior using `main` when testing the new points cursor.

Also, why isn't the zoom/pan function automatically active when using the Select tool? It seems like it should be, at least to my intuition.

**Comment by brisvag** [comment]
Continuing theconversation here from this comment: https://github.com/napari/napari/pull/3733#issuecomment-984620802

@psobolewskiPhD can you add some code? I tried reproducing, but it seems fine to me.

```
import numpy as np
import napari
v = napari.Viewer()
v.add_image(np.random.rand(100, 100))
v.add_points(size=1)
```

https://user-images.githubusercontent.com/23482191/144432021-9dc29f7a-81ce-4b34-a5d1-766d9f8f203c.mp4

**Comment by psobolewskiPhD** [comment]
Using your code everything works.
But using the GUI, click to make a new points layer it doesn't work.
I realize now the key: you use `v.add_points(size=1)`
while the GUI default is `size=10`.
For me your code with:
`v.add_points(size=10)` gives the broken behavior.
The points appear to be ~1 px in size, but in fact they are 10 px so selection is foiled.
Again here I am mousing over the TOP point of the diamond, but the right most is selected and moving.
<img width="1432" alt="image" src="https://user-images.githubusercontent.com/76622105/144433557-96859bc2-1fc5-487d-b5f4-df7985bbc0c8.png">

Edit: if I separate the points >10 px they can be moved reliably. Here the left most point is always reliably moved. The right most and the one below to the left are unreliable.
<img width="1432" alt="image" src="https://user-images.githubusercontent.com/76622105/144433922-0ab0f57d-13d8-4ec0-b182-7cdc48b10fd7.png">

**Comment by psobolewskiPhD** [comment]
I updated my env and the same behavior. New napari info:
```
napari: 0.4.12
Platform: macOS-12.0.1-arm64-arm-64bit
System: MacOS 12.0.1
Python: 3.9.6 | packaged by conda-forge | (default, Jul 6 2021, 08:51:19) [Clang 11.1.0 ]
Qt: 5.15.2
PyQt5: 5.15.4
NumPy: 1.21.0
SciPy: 1.7.0
Dask: 2021.06.2
VisPy: 0.9.4

OpenGL:
- GL version: 2.1 Metal - 76.1
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1680x1050, scale 2.0

Plugins:
- Measurements (skimage regionprops): 0.1.10
- StarDist: 2021.10.25
- Utilities (skimage regionprops): 0.1.10
- aicsimageio-in-memory: 0.4.1
- aicsimageio-out-of-memory: 0.4.1
- animation: 0.0.2
- clEsperanto: 0.10.7
- console: 0.0.4
- napari-tools-menu: 0.1.3
- ome-types: 0.2.6
- scikit-image: 0.4.12
- smo: 2.0.0
- svg: 0.1.5
```

**Comment by brisvag** [comment]
I changed the size to 1 to get closer to your example in hopes of reproducing; with default size, everything is fine as well. However, it seems that your displayed size is not actually 10, cause it should be much bigger (as 10 image pixels, in fact).

As you're on MAC, this just sounds like #2894. Unfortunately, it's a hardware problem :(

**Comment by brisvag** [comment]
More context: vispy/vispy#2078

**Comment by psobolewskiPhD** [comment]
Yup! That's exactly it. I can use @alisterburt code there and reproduce.
and with #2894 this should be closed I guess?

**Comment by brisvag** [comment]
Unfortunately, no. The only solution to this problem is to go deep in vispy and change how Markers are rendered on the deepest level (use hand-made quads rather than the gl points).

**Comment by psobolewskiPhD** [comment]
Ok, what about a kludge where on mac the size is reduced with zoom?
I mean your example `size=1` on my end is perfectly functional both at `astronaut` high zoom. 
In fact, smaller points at high zoom seem more useful!

Otherwise, maybe change the default size to maybe 2 or 3 so the effect is less likely to be triggered? Those are both functional at default zoom for `astronaut` on my end.

Edit: also i meant close the issue since it's sort of a dupe.

**Comment by brisvag** [comment]
Automatically adjusting zoom would require (as you expected) a similar fix as the one required by #3734 . That's not trivial, cause we need access to the camera zoom from the layer, afaik.

Changing default size is not really a solution, since that would depend on what other underlying data is there and how the camera initializes its zoom depending on it... I think the only good solution is to simply rework how points are rendered :/

Closing; further discussion should go to #2894 or vispy/vispy#2078.
